Spanner/src/Transaction.php

```diff
@@ -615,6 +610,14 @@ public function commit(array $options = [])
             throw new \BadMethodCallException('The transaction cannot be committed because it is not active');
         }
 
+        if (is_null($this->transactionId) && isset($this->options['begin'])) {
+            unset($this->options['begin']);
+            $transaction = $this->operation->transaction($this->session, $this->options);
+            if (!is_null($transaction->id())) {
+                $this->setId($transaction->id());
+            }
+        }
```

Comment:
This logic looks a bit unintuitive, could add a comment for why we're doing this (e.g. why do we unset `$transactionId` if `$options['begin'] exists?)

Reply:
Commented!